core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -60,22 +68,22 @@ func NewOutgoingConnectorHandler(gc connector.GatewayConnector, config ServiceCo
 	if !validMethod(method) {
 		return nil, fmt.Errorf("invalid outgoing connector handler method: %s", method)
 	}
-
-	return &OutgoingConnectorHandler{
+	och := &OutgoingConnectorHandler{
 		gc:                  gc,
 		method:              method,
 		responses:           newResponses(),
 		outgoingRateLimiter: outgoingRateLimiter,
 		incomingRateLimiter: incomingRateLimiter,
 		lggr:                lgger,
 		selectorOpts:        opts,
-	}, nil
+	}
+	och.toRetryableFunc = och.toRetryable
+
+	return och, nil
 }
 
 // HandleSingleNodeRequest sends a request to first available gateway node and blocks until response is received
-// TODO: handle retries
 func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context, messageID string, req capabilities.Request) (*api.Message, error) {
-	lggr := logger.With(c.lggr, "messageID", messageID, "workflowID", req.WorkflowID)
 	workflowAllow, globalAllow := c.outgoingRateLimiter.AllowVerbose(req.WorkflowID)
```

Comment:
We need to be careful here: by putting this above the retry logic it implies that any rate limiter would not account for retries, which is probably not what you want